tests/layering/helpers_test.go

```diff
@@ -0,0 +1,611 @@
+package e2e_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"path"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	imagev1 "github.com/openshift/api/image/v1"
+	machineClient "github.com/openshift/client-go/machine/clientset/versioned"
+	"github.com/openshift/machine-config-operator/test/framework"
+	"github.com/openshift/machine-config-operator/test/helpers"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	aggregatedErr "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kubernetes/pkg/credentialprovider"
+)
+
+// Gets the RHCOS base image tag from the BRANCH environment variable or
+// defaults to "latest"; e.g., registry.ci.openshift.org/rhcos-devel/rhel-coreos:<tag>
+func getBaseImageTag() string {
+	branch, found := os.LookupEnv("BRANCH")
+	if !found || branch == "" {
+		return "latest"
+	}
+
+	return strings.ReplaceAll(branch, "release-", "")
+}
+
+// Builds a derived OS image and places it into an Imagesteam. Attempts to use
+// a previously-built image, if one is available.
+func maybeBuildDerivedOSImage(t *testing.T, cs *framework.ClientSet) error {
+	imageStreamConfig := &imagev1.ImageStream{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      imageStreamName,
+			Namespace: mcoNamespace,
+		},
+	}
+
+	_, err := cs.ImageStreams(mcoNamespace).Create(context.TODO(), imageStreamConfig, metav1.CreateOptions{})
+	if err != nil && !k8sErrors.IsAlreadyExists(err) {
+		// If we have an error and it is not an already exists error, something is wrong.
+		return err
+	}
+
+	if k8sErrors.IsAlreadyExists(err) {
+		// We've already got an imagestream matching this name
+		imagestream, err := cs.ImageStreams(mcoNamespace).Get(context.TODO(), imageStreamName, metav1.GetOptions{})
```

Comment:
Not sure how much it matters but I've been asked before to do `ctx := context.TODO()` and then use `ctx` when I'm calling `context.TODO()` repeatedly. You have that here and a few other places

Reply:
What you suggested is good practice and something I should have been doing. Will fix.